### PR TITLE
*: get cluster id directly in pd client

### PR DESCRIPTION
**Pull request by nolouch**
close https://github.com/pingcap/pd/issues/596
PTAL @siddontang @disksing

### CODE REVIEW

File: pkg/apiutil/client.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (c *Client) GetLeader() (*pdpb.Member, error) {
 
 // GetClusterID returns the Cluster ID.
 func (c *Client) GetClusterID() (uint64, error) {
-	clusterURL := c.url + "/cluster"
+	clusterURL := c.url + "/cluster/raft"
```
Comment (siddontang) [inline pkg/apiutil/client.go]: Why do we need to get cluster ID through raft path?

**Comment by siddontang** [comment]
we must distinguish cluster and raft cluster now.

For cluster, it contains TiDB, PD and TiKV.
For raft cluster, it only contains TiKV.

**Comment by siddontang** [comment]
Now for meta.Cluster, I think it should for cluster, not only for raft.

**Comment by siddontang** [comment]
Any test to cover it?

**Comment by siddontang** [comment]
PTAL @disksing

File: server/api/cluster_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	s.cleanup()
+}
+
+func (s *testClusterInfo) TestCluster(c *C) {
+	url := fmt.Sprintf("%s/cluster", s.urlPrefix)
+	c1 := &metapb.Cluster{}
+	err := readJSONWithURL(url, c1)
+	c.Assert(err, IsNil)
+	c2 := &metapb.Cluster{}
+	mustBootstrapCluster(c, s.svr)
```
Comment (siddontang) [inline server/api/cluster_test.go]: no need to bootstrap cluster here, you can change the replica count, and then check the got cluster config again.

**Comment by disksing** [comment]
LGTM

**Pull request merged**